ocaml/xenopsd/lib/xenops_sandbox.ml

```diff
@@ -96,7 +96,19 @@ end = struct
     let gid = base_gid () + domid in
     {root; uid; gid}
 
-  let create chroot paths =
+  let destroy chroot =
+    Xenops_utils.best_effort (Printf.sprintf "removing chroot %s" chroot.root)
+      (fun () -> Xenops_utils.FileFS.rmtree chroot.root
+    )
+
+  let create ~clean chroot paths =
+    if Sys.file_exists chroot.root then
+      if clean then (
+        D.info "%s cleaning up existing chroot %s" __FUNCTION__ chroot.root ;
+        destroy chroot
+      ) else
+        D.info "%s: root %s already exists, continuing anyway" __FUNCTION__
+          chroot.root ;
     try
       Xenops_utils.Unixext.mkdir_rec chroot.root 0o755 ;
```

Comment:
Would this fail if the directory exists? Such that this should be only called when it does not exist?